sos/plugins/juju.py

```diff
@@ -37,85 +19,59 @@ class Juju(Plugin, UbuntuPlugin):
+    # package on a juju machine.
+    files = ('/var/log/juju')
 
     def setup(self):
-        self.add_copy_spec("/var/log/upstart/juju-db.log")
-        self.add_copy_spec("/var/log/upstart/juju-db.log.1")
-        if not self.get_option("all_logs"):
-            # We need this because we want to collect to the limit of all
-            # *.logs in the directory.
-            if(os.path.isdir("/var/log/juju/")):
-                for filename in os.listdir("/var/log/juju/"):
-                    if filename.endswith(".log"):
-                        fullname = os.path.join("/var/log/juju/" + filename)
-                        self.add_copy_spec(fullname)
-            self.add_cmd_output('ls -alRh /var/log/juju*')
-            self.add_cmd_output('ls -alRh /var/lib/juju/*')
+        # Get juju services, they are uniquely named based on the deployment
+        # so we need to parse what their names are.
+        cmd = "systemctl --no-pager --all --no-legend list-units juju\\*"
+        juju_systemctl_output = self.exec_cmd(cmd)["output"]
+        for line in iter(juju_systemctl_output.splitlines()):
+            match = re.match(r"^(juju\S+)\.", line)
+            if match:
+                self.add_journal(units=str(match.group(1)))
+                self.add_service_status(str(match.group(1)))
+
+        # Get agent configs for each agent.
+        agent_configs = []
+        for dirpath, dirnames, filenames in os.walk("/var/lib/juju/agents"):
+            for dirname in dirnames:
+                agent_configs.append(
+                    "/var/lib/juju/agents/" + dirname + "/agent.conf")
+        self.add_copy_spec(agent_configs)
+
+        # Get a directory listing of /var/log/juju and /var/lib/juju
+        self.add_cmd_output(
+            ["ls -alRh /var/log/juju*", "ls -alRh /var/lib/juju*"]
+        )
 
+        if self.get_option("all_logs"):
+            # /var/lib/juju used to be in the default capture moving here
+            # because it usually was way to big.  However, in most cases you
+            # want all logs you want this too.
+            self.add_copy_spec(["/var/log/juju", "/var/lib/juju"])
         else:
-            self.add_copy_spec([
-                "/var/log/juju",
-                "/var/log/juju-*",
-                "/var/lib/juju"
-                # /var/lib/juju used to be in the default capture moving here
-                # because it usually was way to big.  However, in most cases
-                # you want all logs you want this too.
-            ])
-
-        self.add_cmd_output([
-                "juju --version",
-                "juju -v status --format=tabular",
-        ])
-        for service in self.get_deployed_services():
-            self.add_cmd_output([
-                "juju get {}".format(service),
-                "juju get-config {}".format(service),
-                "juju get-constraints {}".format(service)
-            ])
```

Comment:
Are all of these really no longer desired? Is there any command output from `juju` we should be collecting at this point, as this PR is nuking all command collect?

Note that I have absolutely no experience with juju personally so I don't have a horse in this race, but it seems odd to not collect any command output.

Reply:
The redesign will have this running only on machines managed by Juju, they won't even have the juju command present on them.  Yes it  does seem weird, but is fully intentional.

Reply:
Ack.